pkg/agent/openflow/client.go

```diff
@@ -236,6 +236,12 @@ type Client interface {
 	IsIPv4Enabled() bool
 	// Returns if IPv6 is supported on this Node or not.
 	IsIPv6Enabled() bool
+	// GenBasePacketOutBuilder generates a base IP packetOutBuilder, which can have more packet data added.
+	GenBasePacketOutBuilder(srcMAC string, dstMAC string, srcIP string, dstIP string, inPort uint32, outPort uint32) (binding.PacketOutBuilder, error)
+	// SendTCPReject sends TCP reject packet as a packet-out to OVS.
+	SendTCPReject(packetOutBuilder binding.PacketOutBuilder, TCPSrcPort uint16, TCPDstPort uint16, TCPSeqNum uint32, TCPAckNum uint32, isIPv6 bool) error
+	// SendICMPReject sends ICMP reject packet as a packet-out to OVS.
+	SendICMPReject(packetOutBuilder binding.PacketOutBuilder, ICMPData []byte, isIPv6 bool) error
```

Comment:
I think that `SendTCPReset` would be a better method name for the first one.
For the second one, any reason why the method cannot be made generic and support arbitrary ICMP type and code? Once we reach this interface, we are no longer really in the context of the reject implementation.

Reply:
Change both generic.